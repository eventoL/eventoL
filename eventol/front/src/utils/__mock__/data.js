const place1 = {
  address_components: [
    {long_name: '1667', short_name: '1667', types: ['street_number']},
    {long_name: 'Ayacucho', short_name: 'Ayacucho', types: ['route']},
    {
      long_name: 'Centro',
      short_name: 'Centro',
      types: ['neighborhood', 'political'],
    },
    {
      long_name: 'Rosario',
      short_name: 'Rosario',
      types: ['locality', 'political'],
    },
    {
      long_name: 'Departamento Rosario',
      short_name: 'Departamento Rosario',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'Santa Fe',
      short_name: 'Santa Fe',
      types: ['administrative_area_level_1', 'political'],
    },
    {long_name: 'Argentina', short_name: 'AR', types: ['country', 'political']},
    {long_name: 'S2000', short_name: 'S2000', types: ['postal_code']},
    {long_name: 'BYS', short_name: 'BYS', types: ['postal_code_suffix']},
  ],
  formatted_address: 'Ayacucho 1667, S2000BYS Rosario, Santa Fe, Argentina',
  geometry: {
    location: {lat: -32.9586923, lng: -60.629729},
    location_type: 'ROOFTOP',
    viewport: {
      south: -32.9600412802915,
      west: -60.63107798029148,
      north: -32.9573433197085,
      east: -60.628380019708516,
    },
  },
  place_id: 'ChIJ7QvrMwCrt5URpyBml6doLnY',
  plus_code: {
    compound_code: '29RC+G4 Rosario, Santa Fe, Argentina',
    global_code: '47VX29RC+G4',
  },
  types: ['street_address'],
};

const place2 = {
  address_components: [
    {long_name: '1551', short_name: '1551', types: ['street_number']},
    {long_name: 'Sarmiento', short_name: 'Sarmiento', types: ['route']},
    {
      long_name: 'San Nicolás',
      short_name: 'San Nicolás',
      types: ['political', 'sublocality', 'sublocality_level_1'],
    },
    {
      long_name: 'Comuna 1',
      short_name: 'Comuna 1',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'Buenos Aires',
      short_name: 'CABA',
      types: ['administrative_area_level_1', 'political'],
    },
    {long_name: 'Argentina', short_name: 'AR', types: ['country', 'political']},
    {long_name: 'C1042', short_name: 'C1042', types: ['postal_code']},
    {long_name: 'ABC', short_name: 'ABC', types: ['postal_code_suffix']},
  ],
  formatted_address: 'Sarmiento 1551, C1042ABC CABA, Argentina',
  geometry: {
    location: {lat: -34.6052436, lng: -58.388569700000005},
    location_type: 'ROOFTOP',
    viewport: {
      south: -34.6065925802915,
      west: -58.38991868029149,
      north: -34.60389461970851,
      east: -58.387220719708466,
    },
  },
  place_id: 'ChIJbXpzCcTKvJURbeIVAyh6SnU',
  plus_code: {
    compound_code: '9JV6+WH Buenos Aires, Argentina',
    global_code: '48Q39JV6+WH',
  },
  types: ['street_address'],
};

export const event1 = {
  attendees: 10,
  backdrop: 'event1backdrop',
  eventSlug: 'event1slug',
  overview: 'event1overview',
  place: JSON.stringify(place1),
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 1',
  url: 'event1Url',
  id: 1,
};

export const event2 = {
  attendees: 25,
  backdrop: 'event2backdrop',
  eventSlug: 'event2slug',
  overview: 'event2overview',
  place: JSON.stringify(place2),
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 2',
  url: 'event2Url',
  id: 2,
};

export const events = [event1, event2];

export default {
  event1,
  event2,
  events,
};
