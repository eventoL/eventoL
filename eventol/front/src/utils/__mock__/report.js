const getConfirmationReport = (confirmed, notConfirmed) => ({
  confirmed,
  not_confirmed: notConfirmed,
  total: confirmed + notConfirmed,
});

export const usersReports = {
  attendee: {
    with_event_user: getConfirmationReport(1, 2),
    without_event_user: getConfirmationReport(2, 3),
  },
  collaborator: getConfirmationReport(3, 4),
  event_user: getConfirmationReport(4, 5),
  installer: getConfirmationReport(6, 7),
  organizer: getConfirmationReport(7, 8),
  speakers: 9,
};

export const activitiesReport = {
  level_count: {
    '1': 3,
    '2': 2,
    '3': 1,
  },
  status_count: {
    '1': 2,
    '2': 1,
  },
  type_count: {
    Talk: 4,
    Workshop: 2,
  },
  ...getConfirmationReport(1, 2),
};

export const installationsReport = {
  hardware_count: {
    notebook: 1,
    phone: 2,
  },
  software_count: {
    ubuntu: 3,
    fedora: 2,
    arch: 1,
  },
  installer_count: {
    '1': 2,
    '2': 1,
  },
  total: 2,
};

export const eventReport = {
  // users data
  ...usersReports,

  // activities data
  activity: activitiesReport,

  // installations data
  installation: installationsReport,
};

export const event1PrivateData = {
  id: 1,
  email: 'organizer1@mailinator.com',
  organizers: 'FirstName1 LastName1 Username1',
};

export const event2PrivateData = {
  id: 2,
  email: 'organizer2@mailinator.com',
  organizers: 'FirstName2 LastName2 Username2',
};

export const eventsPrivateData = [
  {...event1PrivateData},
  {...event2PrivateData},
];

export const getEventData = (id, name, tags, report, eventSlug, location) => ({
  id,
  name,
  tags,
  report,
  location,
  event_slug: eventSlug,
});

export const event1Data = getEventData(
  1,
  'event name',
  {name: 'tag name', slug: 'tag slug'},
  eventReport,
  'event_slug',
  ['Capital Federal', 'Pompeya', 'Buenos Aires', 'Argentina']
);

export const event2Data = getEventData(
  2,
  'event 2 name',
  {name: 'tag 2 name', slug: 'tag 2 slug'},
  eventReport,
  'event_2_slug',
  ['Capital Federal', 'Monserrat', 'Buenos Aires', 'Argentina']
);

export const eventsData = [{...event1Data}, {...event2Data}];

export const expectedActivityDetail = {
  levels: {'1': 6, '2': 4, '3': 2},
  status: {'1': 4, '2': 2},
  types: {Talk: 8, Workshop: 4},
};

export const expectedInstallationsSoftwares = {arch: 2, fedora: 4, ubuntu: 6};

export const expectedTotals = {
  activities: {
    confirmed: 2,
    details: expectedActivityDetail,
    not_confirmed: 4,
    total: 6,
  },
  attendees: {confirmed: 6, not_confirmed: 10, total: 16},
  collaborators: {confirmed: 6, not_confirmed: 8, total: 14},
  installations: {softwares: expectedInstallationsSoftwares, total: 4},
  installers: {confirmed: 12, not_confirmed: 14, total: 26},
  organizers: {confirmed: 14, not_confirmed: 16, total: 30},
  speakers: 18,
};

export const expectedEventDataWithoutPrivateData = {
  ...event1Data,
  assistanceDetail: {attendees: {confirmed: 3, not_confirmed: 5, total: 8}},
  locationDetail: {
    address: 'Pompeya',
    address_detail: 'Capital Federal',
    province: 'Buenos Aires',
  },
};

export const expectedEventDataWithPrivateData = {
  ...expectedEventDataWithoutPrivateData,
  ...event1PrivateData,
};

export const expectedValues = {
  expectedTotals,
  expectedActivityDetail,
  expectedInstallationsSoftwares,
  expectedEventDataWithPrivateData,
  expectedEventDataWithoutPrivateData,
};

const mockedData = {
  activitiesReport,
  eventData: event1Data,
  event1Data,
  event2Data,
  eventsData,
  eventReport,
  event1PrivateData,
  event2PrivateData,
  eventsPrivateData,
  getEventData,
  installationsReport,
  usersReports,
  expectedValues,
};
export default mockedData;
