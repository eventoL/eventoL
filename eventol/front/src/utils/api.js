import _ from 'lodash';
import Cookies from 'js-cookie';

import Logger from './logger';
import {getReportUrl} from './urls';

export const getCsrf = () => Cookies.get('csrftoken');

export const addQueryString = (url, query) => {
  const queryKeys = query ? Object.keys(query) : [];
  if (!_.isEmpty(queryKeys)) {
    const params = queryKeys.map(k => `${k}=${encodeURIComponent(query[k])}`);
    return `${url}?${params.join('&')}`;
  }
  return url;
};

export const genericFetch = (url, queryString, params) => {
  const newUrl = addQueryString(url, queryString);
  return fetch(newUrl, params)
    .then(res => {
      if (res.ok) {
        return res.json();
      }
      return Promise.reject(res);
    })
    .catch(res => {
      if (res.status === 403) {
        window.location.hash = '/';
        window.location.href = '/';
        Logger.warning('Redirect to /', res.status, res.statusText);
      } else {
        Logger.error('There has been an error', res.status, res.statusText);
      }
    });
};

export const getUrl = (url, queryString) =>
  genericFetch(url, queryString, {
    method: 'GET',
    credentials: 'same-origin',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      'X-CSRFToken': getCsrf(),
    },
  });

export const postUrl = (url, data, queryString) =>
  genericFetch(url, queryString, {
    method: 'POST',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'X-CSRFToken': getCsrf(),
    },
    body: JSON.stringify(data),
    credentials: 'same-origin',
  });

export const loadReports = (pageSize, page, sorted) =>
  getUrl(getReportUrl(pageSize, page, sorted));
