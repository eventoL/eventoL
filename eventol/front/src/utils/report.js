import _ from 'lodash';

export const parseInstallationSoftwares = allData => {
  const softwares = new Set();
  allData.forEach(event =>
    Object.keys(event.report.installation.software_count).forEach(key =>
      softwares.add(key)
    )
  );
  const installationsSoftwares = {};
  softwares.forEach(software => {
    const sum = _.sumBy(
      allData,
      `report.installation.software_count.${software}`
    );
    installationsSoftwares[software] = sum;
  });
  return installationsSoftwares;
};

export const parseActivitiesDetails = allData => {
  const status = new Set();
  const types = new Set();
  const levels = new Set();
  allData.forEach(event => {
    Object.keys(event.report.activity.status_count).forEach(key =>
      status.add(key)
    );
    Object.keys(event.report.activity.type_count).forEach(key =>
      types.add(key)
    );
    Object.keys(event.report.activity.level_count).forEach(key =>
      levels.add(key)
    );
  });
  const activityDetail = {types: {}, status: {}, levels: {}};
  status.forEach(element => {
    const sum = _.sumBy(allData, `report.activity.status_count.${element}`);
    activityDetail.status[element] = sum;
  });
  types.forEach(type => {
    const sum = _.sumBy(allData, `report.activity.type_count.${type}`);
    activityDetail.types[type] = sum;
  });
  levels.forEach(level => {
    const sum = _.sumBy(allData, `report.activity.level_count.${level}`);
    activityDetail.levels[level] = sum;
  });
  return activityDetail;
};

export const parseTotals = allData => {
  const totals = {
    speakers: _.sumBy(allData, 'report.speakers'),
    attendees: {
      confirmed:
        _.sumBy(allData, 'report.attendee.with_event_user.confirmed') +
        _.sumBy(allData, 'report.attendee.without_event_user.confirmed'),
      not_confirmed:
        _.sumBy(allData, 'report.attendee.with_event_user.not_confirmed') +
        _.sumBy(allData, 'report.attendee.without_event_user.not_confirmed'),
      total:
        _.sumBy(allData, 'report.attendee.with_event_user.total') +
        _.sumBy(allData, 'report.attendee.without_event_user.total'),
    },
    organizers: {
      confirmed: _.sumBy(allData, 'report.organizer.confirmed'),
      not_confirmed: _.sumBy(allData, 'report.organizer.not_confirmed'),
      total: _.sumBy(allData, 'report.organizer.total'),
    },
    collaborators: {
      confirmed: _.sumBy(allData, 'report.collaborator.confirmed'),
      not_confirmed: _.sumBy(allData, 'report.collaborator.not_confirmed'),
      total: _.sumBy(allData, 'report.collaborator.total'),
    },
    installers: {
      confirmed: _.sumBy(allData, 'report.installer.confirmed'),
      not_confirmed: _.sumBy(allData, 'report.installer.not_confirmed'),
      total: _.sumBy(allData, 'report.installer.total'),
    },
    activities: {
      confirmed: _.sumBy(allData, 'report.activity.confirmed'),
      not_confirmed: _.sumBy(allData, 'report.activity.not_confirmed'),
      total: _.sumBy(allData, 'report.activity.total'),
      details: parseActivitiesDetails(allData),
    },
    installations: {
      total: _.sumBy(allData, 'report.installation.total'),
      softwares: parseInstallationSoftwares(allData),
    },
  };
  return totals;
};

export const parseEvent = (event, eventsPrivateData = []) => {
  const {
    location,
    report: {attendee},
  } = event;
  const eventData = {
    locationDetail: {
      address_detail: location.slice(0, -3).join(' '),
      address: location[location.length - 3],
      province: location[location.length - 2],
    },
    assistanceDetail: {
      attendees: {
        confirmed:
          attendee.with_event_user.confirmed +
          attendee.without_event_user.confirmed,
        not_confirmed:
          attendee.with_event_user.not_confirmed +
          attendee.without_event_user.not_confirmed,
        total:
          attendee.with_event_user.total + attendee.without_event_user.total,
      },
    },
    ...event,
  };
  if (!_.isEmpty(eventsPrivateData)) {
    const privateData = _.find(eventsPrivateData, {id: event.id});
    if (!_.isEmpty(privateData)) return {...privateData, ...eventData};
  }
  return eventData;
};
