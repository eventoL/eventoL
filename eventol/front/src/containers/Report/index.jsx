import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import Toggle from 'react-input-toggle';

import Logger from '../../utils/logger';
import Title from '../../components/Title';
import Button from '../../components/Button';
import ReportTable from '../../components/ReportTable';
import ExportButton from '../../components/ExportButton';

import {loadReports} from '../../utils/api';
import {parseTotals, parseEvent} from '../../utils/report';

import './react-input-toggle.css';
import 'react-table/react-table.css';

export default class Report extends React.PureComponent {
  static propTypes = {
    communicator: PropTypes.shape({
      addOnMessage: PropTypes.func.isRequired,
    }).isRequired,
    eventsPrivateData: PropTypes.arrayOf(PropTypes.shape()),
  };

  static defaultProps = {
    eventsPrivateData: [],
  };

  state = {
    table: 'confirmed',
    count: 0,
    autoupdate: false,
    data: [],
    totals: {},
    pages: null,
    loading: true,
  };

  componentDidMount() {
    const {communicator} = this.props;
    communicator.addOnMessage(this.updateTable);
    this.loadToggleAutoupdate();
  }

  fetchData = ({pageSize, page, sorted, filtered}) => {
    const {eventsPrivateData} = this.props;
    loadReports(pageSize, page, sorted, filtered)
      .then(({count, results}) => {
        const quotient = Math.floor(count / pageSize);
        const remainder = count % pageSize;
        const pages = remainder > 0 ? quotient + 1 : quotient;
        const data = results.map(event => parseEvent(event, eventsPrivateData));
        const totals = parseTotals(results);
        this.setState({
          count,
          data,
          loading: false,
          pages,
          totals,
        });
      })
      .catch(err => Logger.error(gettext('There has been an error'), err));
  };

  onClick = table => this.setState({table});

  loadToggleAutoupdate = () => {
    let autoupdate = localStorage.getItem('autoupdate');
    if (_.isNull(autoupdate)) autoupdate = false;
    else autoupdate = JSON.parse(autoupdate);
    this.setState({autoupdate});
  };

  handleToggleAutoupdate = () => {
    const {autoupdate} = this.state;
    localStorage.setItem('autoupdate', !autoupdate);
    this.setState({autoupdate: !autoupdate});
  };

  updateTable = () => {
    const {autoupdate} = this.state;
    if (autoupdate) window.location.reload();
  };

  render() {
    const {data, pages, loading, count, table, totals, autoupdate} = this.state;
    const {eventsPrivateData} = this.props;
    return (
      <div>
        <Title label={gettext('National report')}>
          <Toggle
            checked={autoupdate}
            effect="echo"
            label={gettext('Autoupdate')}
            labelPosition="left"
            name={gettext('Autoupdate')}
            onChange={this.handleToggleAutoupdate}
          />
          <Button
            handleOnClick={this.onClick}
            label={gettext('Assistance (confirmed)')}
            name="confirmed"
            type="success"
          />
          <Button
            handleOnClick={this.onClick}
            label={gettext('Assistance detail')}
            name="assitance"
            type="success"
          />
          <Button
            handleOnClick={this.onClick}
            label={gettext('Installations')}
            name="installations"
            type="success"
          />
          <Button
            handleOnClick={this.onClick}
            label={gettext('Activities')}
            name="activities"
            type="success"
          />
          <ExportButton
            ref={exportButton => {
              this.exportButton = exportButton;
            }}
            data={data}
            filename={table}
            label={gettext('Export')}
            type="success"
          />
        </Title>
        <ReportTable
          count={count}
          data={data}
          defaultRows={15}
          eventsPrivateData={eventsPrivateData}
          exportButton={this.exportButton}
          fetchData={this.fetchData}
          isLoading={loading}
          pages={pages}
          table={table}
          totals={totals}
        />
      </div>
    );
  }
}
