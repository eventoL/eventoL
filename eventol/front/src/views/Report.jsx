import React from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';

import getStore from '../store';
import Report from '../containers/Report';
import {getCommunicator} from './utils/communicator';

import './index.scss';

const store = getStore();

window.render_components = properties => {
  window.params = {...properties};
  const communicator = getCommunicator(properties);
  render(
    <Provider store={store}>
      <Report
        communicator={communicator}
        eventsPrivateData={properties.events_private_data}
        user={properties.user}
      />
    </Provider>,
    document.getElementById('root')
  );
};

if (module.hot) {
  if (window.params) window.render_components(window.params);
  module.hot.accept();
}
