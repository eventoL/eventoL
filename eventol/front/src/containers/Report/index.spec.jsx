import wait from 'waait';
import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-input-toggle', () => 'Toggle');

jest.mock('../../components/Title', () => 'Title');
jest.mock('../../components/Button', () => 'Button');
jest.mock('../../components/ReportTable', () => 'TableReport');
jest.mock('../../components/ExportButton', () => 'ExportButton');

jest.mock('../../utils/logger', () => ({error: jest.fn()}));
jest.mock('../../utils/api', () => ({
  loadReports: jest.fn(),
}));
jest.mock('../../utils/report', () => ({
  parseTotals: jest.fn(),
  parseEvent: jest.fn(),
}));

import Toggle from 'react-input-toggle';
import Report from '.';
import {loadReports} from '../../utils/api';
import {events} from '../../utils/__mock__/data';
import {eventsPrivateData} from '../../utils/__mock__/report';

describe('Report', () => {
  let tree;
  let element;
  let instance;
  let component;
  let communicator;

  beforeEach(() => {
    communicator = {
      addOnMessage: jest.fn(),
    };

    element = (
      <Report
        communicator={communicator}
        eventsPrivateData={eventsPrivateData}
      />
    );
    component = renderer.create(element);
    instance = component.root;
    tree = component.toJSON();
  });

  describe('Toggle update autoupdate on onChange event', () => {
    test('toggle autoupdate value', async () => {
      expect(tree).toMatchSnapshot();

      instance.findByType(Toggle).props.onChange();
      component.update(element);
      await wait(0);

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      expect(component.getInstance().state.autoupdate).toBeTruthy();

      instance.findByType(Toggle).props.onChange();
      component.update(element);
      await wait(0);

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      instance = component.getInstance();
      expect(component.getInstance().state.autoupdate).toBeFalsy();
    });
  });

  describe('Snapshots', () => {
    test('Default', async () => {
      expect(tree).toMatchSnapshot();

      component.update(element);
      await wait(0);
      tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('fetch data', async () => {
      loadReports.mockImplementationOnce(() => {
        return new Promise(resolve =>
          resolve({count: events.length, results: events})
        );
      });

      expect(tree).toMatchSnapshot();

      component.update(element);
      await wait(0);
      tree = component.toJSON();

      expect(tree).toMatchSnapshot();

      instance = component.getInstance();
      instance.fetchData(5, 1, false, false);

      component.update(element);
      await wait(0);
      tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
