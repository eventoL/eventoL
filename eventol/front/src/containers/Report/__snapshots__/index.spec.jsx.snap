// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Snapshots Default 1`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Snapshots Default 2`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Snapshots fetchData 1`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Snapshots fetchData 2`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={
        Array [
          "parseEventObject",
          "parseEventObject",
        ]
      }
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={2}
    data={
      Array [
        "parseEventObject",
        "parseEventObject",
      ]
    }
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={false}
    pages={1}
    table="confirmed"
    totals="parseTotalsObject"
  />
</div>
`;

exports[`Report Snapshots loadReports 1`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Snapshots loadReports 2`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Toggle update autoupdate on onChange event toggle autoupdate value 1`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Toggle update autoupdate on onChange event toggle autoupdate value 2`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={true}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;

exports[`Report Toggle update autoupdate on onChange event toggle autoupdate value 3`] = `
<div>
  <Title
    label="National report"
  >
    <Toggle
      checked={false}
      effect="echo"
      label="Autoupdate"
      labelPosition="left"
      name="Autoupdate"
      onChange={[Function]}
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance (confirmed)"
      name="confirmed"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Assistance detail"
      name="assitance"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Installations"
      name="installations"
      type="success"
    />
    <Button
      handleOnClick={[Function]}
      label="Activities"
      name="activities"
      type="success"
    />
    <ExportButton
      data={Array []}
      filename="confirmed"
      label="Export"
      type="success"
    />
  </Title>
  <ReportTable
    count={0}
    data={Array []}
    defaultRows={15}
    eventsPrivateData={
      Array [
        Object {
          "email": "organizer1@mailinator.com",
          "id": 1,
          "organizers": "FirstName1 LastName1 Username1",
        },
        Object {
          "email": "organizer2@mailinator.com",
          "id": 2,
          "organizers": "FirstName2 LastName2 Username2",
        },
      ]
    }
    exportButton={null}
    fetchData={[Function]}
    isLoading={true}
    pages={null}
    table="confirmed"
    totals={Object {}}
  />
</div>
`;
